prusti-tests/tests/verify/pass/arrays/slice-sorted.rs

```diff
@@ -0,0 +1,24 @@
+use prusti_contracts::*;
+
+fn main() {}
+
+
+predicate! {
+    fn sorted(s: &[i32]) -> bool {
+        forall(|i: usize, j: usize| (0 <= i && i < j && j < s.len()) ==> s[i] <= s[j])
+    }
+}
+
+#[requires(sorted(s))]
+fn requires_sorted(s: &[i32]) {
+    if s.len() > 3 {
+        assert!(s[0] <= s[3]);
+    }
+}
+
+/*
+#[requires(sorted(&a[0..2]))]
+fn requires_sorted_start(a: [i32; 3]) {
+    assert!(a[0] <= a[1]);
+}
+*/
```

Comment:
Is this meant to be commented out?

Reply:
yes; the `&a[0..2]` part doesn't work yet (only the unsize-cast that turns a full array into a slice like happens for `arr.len()`).

now that i think about it, i guess implementing full slicing (at least of arrays) makes sense within this PR.

Reply:
alright, within snapshots we can now handle slicing of both arrays and slices :-)